procedural/main.cpp

```diff
@@ -264,8 +274,28 @@ scene_load
 //                 const AtParamValueMap* params, const AtMetadataStore* mds)
 scene_write
 {
+    std::string filenameStr(filename);
+
+    std::string extension = filenameStr.substr(filenameStr.find_last_of('.'));
```

Comment:
We can use `TfGetExtension` here from `pxr/base/tf/pathUtils.h`.

Documentation:
```
/// Returns the extension for a file path
///
/// If \p path is a directory path, an empty path, or a dotfile path, return
/// the empty string. Otherwise return \p path 's dot-separated extension as
/// a string(dot not included).
///
/// Examples:
/// 
/// TfGetExtension('/foo/bar')              -> ''
/// TfGetExtension('/foo/bar/foo.baz')      -> 'baz'
/// TfGetExtension('/foo.bar/foo.baz')      -> 'baz'
/// TfGetExtension('/foo/bar/foo.101.baz')  -> 'baz'
/// TfGetExtension('/foo/bar/.foo.baz')     -> 'baz'
/// TfGetExtension('/foo/bar/.foo')         -> ''
TF_API
std::string TfGetExtension(std::string const& path);
```

For example, just by splitting using the '.' character won't handle a bunch of corner cases, like /foor/bar/.USDA .